Mathlib/GroupTheory/GroupAction/Embedding.lean

```diff
@@ -0,0 +1,67 @@
+/-
+Copyright (c) 2022 Eric Wieser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Wieser
+
+! This file was ported from Lean 3 source module group_theory.group_action.embedding
+! leanprover-community/mathlib commit a437a2499163d85d670479f69f625f461cc5fef9
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.GroupTheory.GroupAction.Group
+import Mathlib.GroupTheory.GroupAction.Pi
+
+/-!
+# Group actions on embeddings
+
+This file provides a `MulAction G (α ↪ β)` instance that agrees with the `MulAction G (α → β)`
+instances defined by `Pi.mulAction`.
+
+Note that unlike the `Pi` instance, this requires `G` to be a group.
+-/
+
+
+universe u v w
+
+variable {G G' α β : Type _}
+
+namespace Function.Embedding
+
+@[to_additive]
+instance smul [Group G] [MulAction G β] : SMul G (α ↪ β) :=
+  ⟨fun g f => f.trans (MulAction.toPerm g).toEmbedding⟩
+
+@[to_additive]
+theorem smul_def [Group G] [MulAction G β] (g : G) (f : α ↪ β) :
+    g • f = f.trans (MulAction.toPerm g).toEmbedding :=
+  rfl
+#align function.embedding.smul_def Function.Embedding.smul_def
```

Comment:
```suggestion
@[to_additive]
theorem smul_def [Group G] [MulAction G β] (g : G) (f : α ↪ β) :
    g • f = f.trans (MulAction.toPerm g).toEmbedding :=
  rfl
#align function.embedding.smul_def Function.Embedding.smul_def
#align function.embedding.vadd_def Function.Embedding.vadd_def
```

Reply:
`to_additive` doesn't add `#align` statements automatically, so these must be added manually throughout the file like this. If you're not sure what `to_additive` did for a particular declaration, you can replace `to_additive` by `to_additive?` and see what it says.